nix/mlir/updated-llvm-lit-cfg-add-libs-to-dylib-path.patch

```diff
@@ -0,0 +1,79 @@
+diff --git a/test/Unit/lit.cfg.py b/test/Unit/lit.cfg.py
+index f15c30dbcdb0..a94de8bb145e 100644
+--- a/test/Unit/lit.cfg.py
++++ b/test/Unit/lit.cfg.py
+@@ -3,6 +3,7 @@
+ # Configuration file for the 'lit' test runner.
+ 
+ import os
++import platform
+ import subprocess
+ 
+ import lit.formats
+diff --git a/test/lit.cfg.py b/test/lit.cfg.py
+index 7a1e30b49b0c..ec9c724202fd 100644
+--- a/test/lit.cfg.py
++++ b/test/lit.cfg.py
+@@ -41,6 +41,26 @@ llvm_config.with_environment("PATH", config.llvm_tools_dir, append_path=True)
+ # Propagate some variables from the host environment.
+ llvm_config.with_system_environment(["HOME", "INCLUDE", "LIB", "TMP", "TEMP"])
+ 
++# Add the LLVM dynamic libs to the platform-specific loader search path env var:
++#
++# TODO: this is copied from `clang`'s `lit.cfg.py`; should unify..
++def find_shlibpath_var():
++    if platform.system() in ['Linux', 'FreeBSD', 'NetBSD', 'OpenBSD', 'SunOS']:
++        yield 'LD_LIBRARY_PATH'
```

Comment:
Why is this written as a generator? We could move the `else` branch below into a catch all case and avoid the for loop

Reply:
I don't disagree but this is how it came from upstream (nixpkgs/pkgs/development/compilers/llvm/git/llvm/llvm-lit-cfg-add-libs-to-dylib-path.patch). I think it's best to leave it as is.